suite-common/suite-sync/src/storage/tests/refreshSuiteSyncKeys.test.ts

```diff
@@ -0,0 +1,185 @@
+    >(),
+    ensureDelegatedIdentityKey: jest.fn<
+        ReturnType<EnsureDelegatedIdentityKey>,
+        Parameters<EnsureDelegatedIdentityKey>
+    >(),
+});
+
+const createDevice = (
+    overrides: Partial<TrezorDevice> = {},
+    deviceStateOverrides?: Partial<TrezorDevice["state"]>,
+): TrezorDevice =>
+    ({
+        id: 'test-device-id',
+        connected: true,
+        mode: 'normal',
+        ...(stateOverrides
+            ? {
+                  state: {
+                      device: {
+                          path: 'test-path',
+                          instance: 1,
+                      },
+                      staticSessionId: 'test-session-id',
+                      ...stateOverrides,
+                  },
+              }
+            : {}),
+        ...overrides,
+    }) as unknown as TrezorDevice;
+
+describe(refreshSuiteSyncKeys.name, () => {
+    it('fails with "unable to get keys" when device is not initialized', async () => {
+        const deps = createDeps();
+        const refreshSuiteSyncKeys = createRefreshSuiteSync(deps);
+        const result = await refreshSuiteSyncKeys({
+            device: createDevice(),
+        });
+
+        expect(result.ok).toEqual(false);
+        if (!result.ok) {
+            expect(result.error.type).toBe('RefreshSuiteKeysUnavailable');
+        }
+    });
+
+    it('returns an suite sync owner when device has state and is available', async () => {
+        const mockOwner = {
+            ownerId: asSuiteSyncOwnerId('test-owner-id'),
+            ownerSecret: asSuiteSyncOwnerSecretHex('test-owner-secret'),
+        };
+        const deps = createDeps();
+        deps.loadSuiteSyncOwnerFromState.mockResolvedValue(mockOwner);
+
+        const refreshSuiteSyncKeys = createRefreshSuiteSync(deps);
+        const result = await refreshSuiteSyncKeys({
+            device: createDevice({}, {}),
+        });
+
+        expect(result.ok).toEqual(true);
+        if (result.ok) {
+            expect(result.value).toEqual(mockOwner);
+        }
+    });
+
+    it('requests refreshing of the keys when they arent available in the state', async () => {
+        const ensuredOwner = {
+            ownerId: asSuiteSyncOwnerId('new-owner-id'),
+            ownerSecret: asSuiteSyncOwnerSecretHex('new-owner-secret'),
+        };
+        const deps = createDeps();
+        deps.loadSuiteSyncOwnerFromState.mockResolvedValue(null);
+        deps.ensureSuiteSyncOwner.mockResolvedValue(ok(ensuredOwner));
+        deps.ensureDelegatedIdentityKey.mockResolvedValue(
+            ok(asDelegatedIdentityKey('delegated-key-value')),
+        );
+
+        const refreshSuiteSyncKeys = createRefreshSuiteSync(deps);
+        await refreshSuiteSyncKeys({
+            device: createDevice({}, {}),
+        });
+
+        expect(deps.ensureDelegatedIdentityKey).toHaveBeenCalled();
+    });
+
+    it('returns refresh unavailable when device is disconnected', async () => {
```

Comment:
```suggestion
    it('fails to get keys, when device is disconnected', async () => {
```

You shall go for `returns` as only last resort